packages/edit-post/src/components/layout/use-should-iframe.js

```diff
@@ -5,36 +5,29 @@ import { store as editorStore } from '@wordpress/editor';
 import { useSelect } from '@wordpress/data';
 import { store as blocksStore } from '@wordpress/blocks';
 
-/**
- * Internal dependencies
- */
-import { store as editPostStore } from '../../store';
-
 const isGutenbergPlugin = globalThis.IS_GUTENBERG_PLUGIN ? true : false;
 
 export function useShouldIframe() {
-	const {
-		isBlockBasedTheme,
-		hasV3BlocksOnly,
-		isEditingTemplate,
-		hasMetaBoxes,
-	} = useSelect( ( select ) => {
-		const { getEditorSettings, getCurrentPostType } = select( editorStore );
-		const { getBlockTypes } = select( blocksStore );
-		const editorSettings = getEditorSettings();
-		return {
-			isBlockBasedTheme: editorSettings.__unstableIsBlockBasedTheme,
-			hasV3BlocksOnly: getBlockTypes().every( ( type ) => {
-				return type.apiVersion >= 3;
-			} ),
-			isEditingTemplate: getCurrentPostType() === 'wp_template',
-			hasMetaBoxes: select( editPostStore ).hasMetaBoxes(),
-		};
-	}, [] );
+	const { isBlockBasedTheme, hasV3BlocksOnly, isEditingTemplate } = useSelect(
+		( select ) => {
+			const { getEditorSettings, getCurrentPostType } =
+				select( editorStore );
+			const { getBlockTypes } = select( blocksStore );
+			const editorSettings = getEditorSettings();
+			return {
+				isBlockBasedTheme: editorSettings.__unstableIsBlockBasedTheme,
+				hasV3BlocksOnly: getBlockTypes().every( ( type ) => {
+					return type.apiVersion >= 3;
+				} ),
+				isEditingTemplate: getCurrentPostType() === 'wp_template',
+			};
+		},
+		[]
+	);
 
 	return (
-		( ( hasV3BlocksOnly || ( isGutenbergPlugin && isBlockBasedTheme ) ) &&
-			! hasMetaBoxes ) ||
+		hasV3BlocksOnly ||
+		( isGutenbergPlugin && isBlockBasedTheme ) ||
```

Comment:
Side note: I think we should remove the special treatment for the Gutenberg plugin. The different behavior from the WP core sometimes results in discovering no-framed editor bugs after the code has shipped.

Sorry, I can't remember examples now, but having the same condition in both builds makes sense, IMO.

Reply:
I was thinking the same as I was reviewing this. The conditional here seems odd at this point

Reply:
We are discussing this, I want to formally remove it in a separate PR.

Reply:
100%. My comment was just a side note.